pkg/kubecfg/vm_test.go

```diff
@@ -0,0 +1,89 @@
+package kubecfg
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/kubecfg/kubecfg/utils"
+)
+
+func tempFile(t *testing.T, body string) (filename string) {
+	t.Helper()
+
+	tmpfile, err := os.CreateTemp("", "kubecfg-test-*.jsonnet")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := io.WriteString(tmpfile, body); err != nil {
+		tmpfile.Close()
+		t.Fatal(err)
+	}
+	if err := tmpfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		os.Remove(tmpfile.Name())
+	})
```

Comment:
```suggestion
	tmpfile, err := os.CreateTemp(t.TempDir(), "kubecfg-test-*.jsonnet")
	if err != nil {
		t.Fatal(err)
	}
	if _, err := io.WriteString(tmpfile, body); err != nil {
		tmpfile.Close()
		t.Fatal(err)
	}
	if err := tmpfile.Close(); err != nil {
		t.Fatal(err)
	}
```

Thoughts on doing something like this?

I.e. we pass in [`t.TempDir()`](https://pkg.go.dev/testing#T.TempDir) which is from the `testing` stdlib that will automatically provide a directory that will be cleaned up after the test anyway - allowing us to slightly simplify the setup and remove our own call to `t.Cleanup`.


> TempDir returns a temporary directory for the test to use. The directory is automatically removed by Cleanup when the test and all its subtests complete.

Reply:
Ah yeah I forgot about that helper